docs/scos/dev/front-end-development/202307.0/oryx/building-applications/styling/oryx-responsive-design.md

```diff
@@ -0,0 +1,215 @@
+
+In order to associate styles to a specific screen size, you can link the styles in the component definition. The styles can be lazily loaded, so that irrelevant styles are not loaded.
+
+<!-- link "component definition" to the component docs once published -->
+
+```ts
+export const selectComponent = componentDef({
+  name: "oryx-select",
+  // ...
+  stylesheets: [
+    {
+      rules: () => import("./styles").then((m) => m.screenStyles),
+    },
+  ],
+});
+```
+
+The example above, loads a file (`styles.ts`), that exports `const screenStyles`. The rules are expected to be an array of `CssStylesWithMedia` type, which is exported from the [utilities package](https://www.npmjs.com/package/@spryker-oryx/utilities). The lazy loaded styles can contain either the styles for a single screen size, or various screen sizes.
+
+```ts
+const screenStyles: CssStylesWithMedia[] = [
+  {
+    media: {
+      [DefaultMedia.Screen]: Breakpoint.Sm,
+    },
+    css: css`
+      :host {
+        // put styles for small screens here
+      }
+    `,
+  },
+];
+```
+
+{% info_block infoBox "" %}
+
+When you do not need to align your custom styles with configurable breakpoints in your custom code, as the breakpoints are no likely to change, you can avoid the configurable styles in the component definition and use standard media queries in your stylesheets.
+
+{% endinfo_block %}
+
+## Responsive Layouts
+
+The layout system has build in support for responsive layouts, which are driven by [design tokens](/docs/scos/dev/front-end-development/{{page.version}}/oryx/building-applications/styling/oryx-design-tokens.html). Layouts, such as a grid or carousel are designed to render different number of items per row or column, depending on the screen size. The number of items is configurable as a design token.
+
+Another important concept for responsive design is the layout container and the page bleed. The layout container has a maximum width, to avoid very wide layouts on users with a large screen. The container size and minimum bleed are configurable as design tokens.
+
+All design tokens are configurable per screen size, which allows you to control the layout per screen, without changing component styles.
+
+The layout system is covered in more detail in the layout documentation.
+
+## Responsive Visibility Rules
+
+Some experiences require an alternative page structure, which means that some components should or should not be rendered in a specific screen size. The Experience Data that you setup for creating [Pages](/docs/scos/dev/front-end-development/{{page.version}}/oryx/building-pages/oryx-pages.html) and [compositions](/docs/scos/dev/front-end-development/{{page.version}}/oryx/building-pages/oryx-compositions.html) allows you to define visibility rules. The visibility rules can be configured to hide a composition or component for a specific screen size.
+
+The following example shows you how to create a simple visibility rule for a specific screen.
```

Comment:
Do we need to mention here `hideByRule` visibility option with short description?

Reply:
I've left it out intentionally, as it must be covered by a dedicated doc on component visibility. We do not have such doc yet, which is one of the reasons I added the high-level section here. I just don't want to cover the full doc here.